src/core/lib/channel/channel_stack_builder.cc

```diff
@@ -25,252 +25,59 @@
-}
-
-bool grpc_channel_stack_builder_move_prev(
-    grpc_channel_stack_builder_iterator* iterator) {
-  if (iterator->node == &iterator->builder->begin) return false;
-  iterator->node = iterator->node->prev;
-  return true;
-}
+namespace grpc_core {
 
-grpc_channel_stack_builder_iterator* grpc_channel_stack_builder_iterator_find(
-    grpc_channel_stack_builder* builder, const char* filter_name) {
-  GPR_ASSERT(filter_name != nullptr);
-  grpc_channel_stack_builder_iterator* it =
-      grpc_channel_stack_builder_create_iterator_at_first(builder);
-  while (grpc_channel_stack_builder_move_next(it)) {
-    if (grpc_channel_stack_builder_iterator_is_end(it)) break;
-    const char* filter_name_at_it =
-        grpc_channel_stack_builder_iterator_filter_name(it);
-    if (strcmp(filter_name, filter_name_at_it) == 0) break;
+ChannelStackBuilder::~ChannelStackBuilder() {
+  if (args_ != nullptr) {
+    grpc_channel_args_destroy(args_);
   }
-  return it;
 }
 
-bool grpc_channel_stack_builder_move_prev(
-    grpc_channel_stack_builder_iterator* iterator);
-
-void grpc_channel_stack_builder_set_channel_arguments(
-    grpc_channel_stack_builder* builder, const grpc_channel_args* args) {
-  if (builder->args != nullptr) {
-    grpc_channel_args_destroy(builder->args);
+ChannelStackBuilder& ChannelStackBuilder::SetTarget(const char* target) {
+  if (target == nullptr) {
+    target_.clear();
+  } else {
+    target_ = target;
   }
-  builder->args = grpc_channel_args_copy(args);
-}
-
-const grpc_channel_args* grpc_channel_stack_builder_get_channel_arguments(
-    grpc_channel_stack_builder* builder) {
-  return builder->args;
+  return *this;
 }
 
-void grpc_channel_stack_builder_set_transport(
-    grpc_channel_stack_builder* builder, grpc_transport* transport) {
-  GPR_ASSERT(builder->transport == nullptr);
-  builder->transport = transport;
-}
-
-grpc_transport* grpc_channel_stack_builder_get_transport(
-    grpc_channel_stack_builder* builder) {
-  return builder->transport;
-}
-
-bool grpc_channel_stack_builder_append_filter(
-    grpc_channel_stack_builder* builder, const grpc_channel_filter* filter,
-    grpc_post_filter_create_init_func post_init_func, void* user_data) {
-  grpc_channel_stack_builder_iterator* it =
-      grpc_channel_stack_builder_create_iterator_at_last(builder);
-  bool ok = grpc_channel_stack_builder_add_filter_before(
-      it, filter, post_init_func, user_data);
-  grpc_channel_stack_builder_iterator_destroy(it);
-  return ok;
-}
-
-bool grpc_channel_stack_builder_remove_filter(
-    grpc_channel_stack_builder* builder, const char* filter_name) {
-  grpc_channel_stack_builder_iterator* it =
-      grpc_channel_stack_builder_iterator_find(builder, filter_name);
-  if (grpc_channel_stack_builder_iterator_is_end(it)) {
-    grpc_channel_stack_builder_iterator_destroy(it);
-    return false;
+ChannelStackBuilder& ChannelStackBuilder::SetChannelArgs(
+    const grpc_channel_args* args) {
+  if (args_ != nullptr) {
```

Comment:
Same here: No need to check this, since it's already checked inside of `grpc_channel_args_destroy()`.

Reply:
done